### PR TITLE
Update HTTP references to HTTPS

### PR DESCRIPTION
**Pull request by mdawaffe**
#### Changes proposed in this Pull Request:

It's much easier to see what's going on if you look at this commit by commit.

* HTTPS links in comments.
* Clickable HTTPS links that do not require string changes.
* Clickable HTTPS links that do require string changes.
* HTTPS for shortcodes and embeds.
* Some functional changes.

Only the last two need real testing.

That there are a large number of HTTP/legacy resources referenced in Jetpack was pointed out to us by @forced-request. Thanks for helping us clean this up!

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
No: bug fixes.

#### Testing instructions:
* Does Jetpack Comments still work? Submit a new comment.
* Does the notifications UI still work?
* Most everything else is trivial.

#### Proposed changelog entry for your changes:
* Use HTTPS URLs when linking to external sites when possible.

### CODE REVIEW

**Review by jeherve — DISMISSED** [review]
This looks good to me. 👍

File: modules/theme-tools/site-logo.php [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?php
 /*
  * Site Logo.
- * @see http://jetpack.com/support/site-logo/
+ * @see https://jetpack.com/support/site-logo/
```
Comment (simison) [inline modules/theme-tools/site-logo.php]: Seems to just redirect to https://jetpack.com/features/design/

**Review by jeherve — DISMISSED** [review]
Approving again now that the tests pass.

**Review by lezama — APPROVED** [review]

**Comment by mdawaffe** [comment]
Let's get this in before there are more conflicts :)

**Pull request merged**